### DISCUSSION TITLE
MongoDB - Removing Object within Nested Array

### DISCUSSION THREAD

**Discussion by JayP718** [Budibase/budibase#9539] [Help] [answered]
Hi,

I was able to generate a query which pushes an object to a nested array.

However, now I'm unable to generate a query to remove the object from that nested array.

{ "_id": {"$eq":"ObjectId('{{set_id}}')"} },
 {"$pull": {"tax_records": {"tx_pid": {{tx_pid}}  } } }

I receive the error "Update Document requires Atomic Operators"

**Comment by melohagan** [comment, marked as answer]
Hey @JayP718 

Just to double check, does this query work in Compass or through the CLI? Does the query work if you replace the bindings with manual values?

**Reply by JayP718** [reply]
Hi Mel so this query does work within the CLI.

Here's how it looks within the CLI and hard coding values.

Atlas atlas-b5qeol-shard-0 [primary] research> db.sets.updateOne({ _id : ObjectId("63e00f87fa84211bcf43bd60") },{$pull : {"tax_records" : { "txpid" : "017-986-206"}}})


{
  acknowledged: true,
  insertedId: null,
  matchedCount: 1,
  modifiedCount: 1,
  upsertedCount: 0
}

**Reply by JayP718** [reply]
Hi @melohagan ,

Thanks again after some more debugging I was able to figure it out.

Best,

Jay

**Reply by JayP718** [reply]
It was an issue with the quotes. Here's the query if anyone else has a similar issue 

{ "_id":"ObjectId('{{sets_id}}')" },{"$pull" : {"mls_records" : { "id_coumn" : {{id_coumn}} }}}

you just want to be sure quotes are included or excluded depending on the field type within mongodb
